### PR TITLE
Local task executors in registry controller

### PR DESCRIPTION
**Pull request by shrutiparabgoogle**
Summary of my changes:

* Added local task executors which are essentially go routines executing `os/exec` commands.
* Added test for `controller-update_test.go` which is end to end in nature. The difference between `controller_test.go` and this one is, `controller` tests check for the accuracy of the actions generated whereas, `controller-update` test executes the command end to end and checks for the existence of artifacts which were generated as a result of execution of the actions.
* Addition of `controller-update` tests with the same `controller-demo` test project made the `make test` command flaky. I was able to fix it by keeping the projects separate for the two tests. Hence, you will see that I have renamed the project `controller-demo` to `controller-test` in file `controller_test.go`.

### CODE REVIEW

**Comment by shrutiparabgoogle** [comment]
@timburks for reviewing the task executors implementation, actions generation and execution will happen in different stages as we discussed
@seaneganx for reviewing cobra command test implementation and general golang conventions.

**Review by seaneganx — APPROVED** [review]

**Comment by seaneganx** [comment]
One other thing I noticed is the file naming with hyphens instead of underscores. I haven't seen hyphens in filenames before but I also don't see any official guidance on the subject. Is there a reason for using this name style for the new files and updating manifest_reader to manifest-reader?

**Comment by timburks** [comment]
LGTM

As you noticed, tests run concurrently, so tests generally need to run in their own projects.

**Review by timburks — APPROVED** [review]

File: cmd/registry/controller/local-tasks.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"os/exec"
+	"log"
+	"strings"
+)
+
+
+type ExecCommandTask struct {
+	Action string
+	TaskId string
```
[the remark this answers is not in the document]
Comment (shrutiparabgoogle) [reply cmd/registry/controller/local-tasks.go]: Done

File: cmd/registry/controller/local-tasks.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	if err != nil {
+		return err
+	}
+	log.Printf("Finished executing taskId %s with action: %s\n Output: %s", task.TaskId, task.Action, output)
+	return nil
+}
+
+func (task *ExecCommandTask) GetAction() string {
+	return task.Action
+}
```
[the remark this answers is not in the document]
Comment (shrutiparabgoogle) [reply cmd/registry/controller/local-tasks.go]: Thanks for noticing this. The struct fields were initially private, missed taking this one out. 

Done

File: cmd/registry/cmd/controller-update_test.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	// Call the controller update command
+	args := append([]string{"controller", "update"}, "../controller/test/manifest_e2e.yaml")
+	rootCmd.SetArgs(args)
+	if err = controllerUpdateCmd.Execute(); err != nil {
+		t.Fatalf("Execute() with args %v returned error: %s", args, err)
+	}
+
+
+	// List all the artifacts
+	got := &[]string{}
```
[the remark this answers is not in the document]
Comment (shrutiparabgoogle) [reply cmd/registry/cmd/controller-update_test.go]: Done.

Thanks for that extra note😃  Will keep in mind.

File: cmd/registry/cmd/controller-update_test.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+			Availability: "GENERAL",
+		},
+	})
+	if err != nil {
+		t.Fatalf("Failed CreateApi %s: %s", apiID, err.Error())
+	}
+
+	// Create Versions 1.0.0, 1.0.1, 1.1.0
+	_, err = client.CreateApiVersion(ctx, &rpc.CreateApiVersionRequest{
+		Parent:       "projects/" + testProject + "/apis/" + apiID,
```
[the remark this answers is not in the document]
Comment (shrutiparabgoogle) [reply cmd/registry/cmd/controller-update_test.go]: Done.

**Comment by shrutiparabgoogle** [comment]
> One other thing I noticed is the file naming with hyphens instead of underscores. I haven't seen hyphens in filenames before but I also don't see any official guidance on the subject. Is there a reason for using this name style for the new files and updating manifest_reader to manifest-reader?

The change in the filename is to be consistent with the `cmd/registry` directory. I noticed the filenames in this directory are all hyphens except for test files with `_test` extension.

**Pull request merged**